core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/store/dot-experiments-reports-store.ts

```diff
@@ -100,33 +103,39 @@ export class DotExperimentsReportsStore extends ComponentStore<DotExperimentsRep
         experiment?.trafficProportion?.variants.find(({ promoted }) => promoted)
     );
 
-    readonly hasEnoughSessions$: Observable<boolean> = this.select(
-        ({ results }) => results != null && results.sessions.total > 0
+    // Bayesian Chart
+    readonly getBayesianChartData$ = this.select(({ results }) =>
+        results ? { datasets: getBayesianDatasets(results) } : null
     );
 
-    readonly setComponentStatus = this.updater(
-        (state: DotExperimentsReportsState, status: ComponentStatus) => ({
-            ...state,
-            status
-        })
+    readonly hasEnoughDataForShowBayesianChart$: Observable<boolean> = this.select(
+        this.state$,
+        this.getBayesianChartData$,
+        ({ results }, chartData) => {
+            if (results && chartData) {
+                const { datasets } = chartData;
+
+                return (
+                    results.bayesianResult.suggestedWinner != BayesianStatusResponse.NONE &&
+                    datasets.every((dataset) => dataset.data.length > 0)
+                );
+            }
+
+            return false;
+        }
     );
 
-    readonly setExperiment = this.updater(
-        (state: DotExperimentsReportsState, experiment: DotExperiment) => ({
-            ...state,
-            experiment: {
-                ...state.experiment,
-                ...experiment
-            }
-        })
+    readonly bayesianChart$: Observable<DotChart> = this.select(
+        this.getBayesianChartData$,
+        this.hasEnoughDataForShowBayesianChart$,
+        (chartData, hasEnoughData) => {
+            return { hasEnoughData, chartData };
+        }
     );
 
-    readonly showExperimentSummary$: Observable<boolean> = this.select(({ experiment }) =>
-        Object.values([
-            DotExperimentStatus.ENDED,
-            DotExperimentStatus.RUNNING,
-            DotExperimentStatus.ARCHIVED
-        ]).includes(experiment?.status)
+    // Daily Chart
+    readonly hasEnoughSessions$: Observable<boolean> = this.select(
+        ({ results }) => results != null && results.sessions.total >= MINIMUM_SESSIONS_TO_SHOW_CHART
```

Comment:
The assertion to null is not needed `result && ...`

Reply:
done